### PR TITLE
Fix broken links to subsections of the release note

### LINKED ISSUE #6638: Links to subsections of the release note don't work
## Description
For example https://ballerina.io/downloads/swan-lake-release-notes/swan-lake-2201.2.0#http-package-1 for Standard library updates > Improvements > http package

### PR DESCRIPTION
**Pull request by AnsarMahir**
This fix resolves an issue where the page wasn't scrolling to subsections with suffixed IDs (e.g., #new-features-1) when accessed directly via URL. Did some modularization of code to clarity and

> Fixes #6638 

## Checklist

- [ ] **Page addition**
  - [ ] Add `permalink` to pages.

- [ ] **Page removal**
  - [ ] Remove entry from corresponding left nav YAML file.
  - [ ] Add `redirect_from` on the alternative page.
  - [ ] If no alternative page, add redirection on the `redirections.js ` file.

- [ ] **Page rename**
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

- [ ] **Page restrcuture**
  - [ ] Add `permalink` to pages.
  - [ ] Add front-matter `redirect_from`.
  - [ ] Add front-matter `redirect_to:` (if applicable).

### CODE REVIEW

**Comment by sm1990** [comment]
@AnsarMahir Commit messages should be in imperative mood. See https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md#submit-your-contribution

**Comment by sm1990** [comment]
@AnsarMahir The correct heading is not getting highlighted in the toc. See the attached video. 

https://github.com/user-attachments/assets/05a11a22-b53a-4239-a0f8-3b454ecdb8c7

**Comment by AnsarMahir** [comment]
@sm1990 Okay. So I believe this issue of highlighting happens because when fetching from the URL, there is a gap between the top edge visual area and the title. but when using TOC, clicking a title will make the title as the top element in the viewable area. So i have now fixed it. 

But there is a fundamental underlying issue, when it comes to subsection topics, without using TOC , if we just scroll through the website, these topics only get highlighted in TOC after they disappear from the viewable area. I have attached a video for reference regarding this. Due to this issue, the highlighted element is one element behind the actual one. (You can check this on the live website also)

https://github.com/user-attachments/assets/1395f02e-77d6-4f21-9faa-89ec29f28448

**Comment by AnsarMahir** [comment]
@sm1990 fixed that TOC highlighting issue also. let me know your thoughts

File: components/common/pg-toc/Toc.js [hunk 1]
```diff
@@ -211,14 +290,12 @@ export default function Toc(props) {
...
+            className={`title-anchor ${item.level}${item.active ? " active" : ""}`}
             onClick={(e) => clickMe(e.target, item.sectionId, item.unique)}
             dangerouslySetInnerHTML={{ __html: item.text }}
           />
         ))}
       </div>
     </>
   );
-}
+}
```
Comment (sm1990) [inline components/common/pg-toc/Toc.js:301]:
```suggestion
}

```

**Review by sm1990 — APPROVED** [review]

**Comment by sm1990** [comment]
@AnsarMahir Thank you for your contribution. Please fill the form and follow the instructions in https://ballerina.io/hacktoberfest/#rules to claim your credits. :smile:

**Pull request merged**